isotovideo

```diff
@@ -386,133 +395,185 @@ sub check_asserted_screen {
-            last;
-        }
-        if ($r == $bmwqemu::backend->{from_child}) {
-            myjsonrpc::send_json($backend_requester, {ret => $rsp->{rsp}});
-            $backend_requester = undef;
-            next;
-        }
-        if ($rsp->{cmd} =~ m/^backend_(.*)/) {
-            die "we need to implement a backend queue" if $backend_requester;
-            $backend_requester = $r;
-            my $cmd = $1;
-            delete $rsp->{cmd};
-            myjsonrpc::send_json($bmwqemu::backend->{to_child}, {cmd => $cmd, arguments => $rsp});
-            next;
-        }
-        if ($rsp->{cmd} eq 'set_current_test') {
-            $bmwqemu::backend->_send_json({cmd => 'set_serial_offset'});
-            $current_test_name = $rsp->{name};
-            myjsonrpc::send_json($r, {ret => 1});
-            next;
-        }
-        if ($rsp->{cmd} eq 'tests_done') {
-            $r         = $rsp->{died};
-            $completed = $rsp->{completed};
-            CORE::close($testfd);
-            $testfd = undef;
-            kill_autotest;
-            $loop = 0;
-            next;
+    my ($io_handle, $rsp) = @_;
+    my $cmd = $rsp->{cmd};
+
+    # write debugging info
+    myjsonrpc::send_json($cmd_srv_fd, {paused => $rsp});
+    print("isotovideo: paused, so not passing $cmd to backend\n");
+
+    # postpone execution of command
+    $postponed_io_handle = $io_handle;
+    $postponed_command   = $rsp;
+
+    # send no reply to autotest, just let it wait
+    return 1;
+}
+
+sub process_command {
+    my ($io_handle, $rsp) = @_;
+
+    if ($rsp->{cmd} =~ m/^backend_(.*)/) {
+        return if handle_paused_test_execution($io_handle, $rsp);
+        die 'isotovideo: we need to implement a backend queue' if $backend_requester;
+        $backend_requester = $io_handle;
+        my $cmd = $1;
+        delete $rsp->{cmd};
+
+        # send debugging info for ws clients
+        myjsonrpc::send_json($cmd_srv_fd, {$cmd => $rsp});
+        # pass command to backend
+        myjsonrpc::send_json($bmwqemu::backend->{to_child}, {cmd => $cmd, arguments => $rsp});
+        return;
+    }
+    if ($rsp->{cmd} eq 'set_pause_at_test') {
+        $pause_test_name = $rsp->{name};
+
+        # print/send debugging info
+        print("isotovideo: test execution will be paused at test $pause_test_name\n");
+        myjsonrpc::send_json($cmd_srv_fd, {set_pause_at_test => $pause_test_name});
+
+        myjsonrpc::send_json($io_handle, {ret => 1});
+        return;
+    }
+    if ($rsp->{cmd} eq 'resume_test_execution') {
+        # print/send debug info
+        print($stop_passing_commands_to_backend ?
+              "isotovideo: test execution will be resumed\n"
+            : "isotovideo: resuming test execution requested but not paused anyways\n"
+        );
+        myjsonrpc::send_json($cmd_srv_fd, {resume_test_execution => $postponed_command});
+
+        # no longer stop passing commands to backend
+        $stop_passing_commands_to_backend = 0;
```

Comment:
How about renaming it to `pass_commands_to_backend` it's a bit less confusing when reading sentences/comments like this.

Reply:
Mh... I made this negative because if the variable is set, that's the special case.